### PR TITLE
feat: share image/username on separate topic on a community

### LINKED ISSUE #2707: Share image/username on separate topic on a community
Users should subscribe to a separate topic when they join a community and they should periodically post their profile picture/ens names etc, in the same way as we do for their personal contact code topic.

### Implementation

I think most of it should be already implemented, we just need to make sure a user subscribes to that topic when joining a community, and that it publishes on it, in `publishContactCode`

### PR DESCRIPTION
**Pull request by osmaczko**
- `publishContactCode` now also publishes on joined communities separate topic
- `publishIdentityImage` is now triggered each time user joins a new community, this is to ensure other community members got updated with user details immediately

closes: #2707

### CODE REVIEW

**Review by 0x-r4bbit — APPROVED** [review]
Changes LGTM.
Can we have 1-2 tests for this please?

**Review by richard-ramos — APPROVED** [review]

**Comment by osmaczko** [comment]
> Changes LGTM. Can we have 1-2 tests for this please?

@PascalPrecht tests added, please take a look: https://github.com/status-im/status-go/pull/2745/files#diff-d4f4b1d3dcfd1f9dda7f6d7a397650c30b499042a7770a1df9a6fe80d61f04fe

File: protocol/communities_messenger_test.go [hunk 1]
```diff
@@ -390,6 +399,92 @@ func (s *MessengerCommunitiesSuite) TestJoinCommunity() {
...
+		})
+		s.Require().NoError(err)
+
+		// Join the community
+		response, err = user.JoinCommunity(context.Background(), community.ID())
+		s.Require().NoError(err)
+		s.Require().NotNil(response)
+	}
+
+	patryk := s.newMessenger()
```
Comment (cammellos) [inline protocol/communities_messenger_test.go:463]: Bob's going to be jealous :sunglasses:

**Review by cammellos — APPROVED** [review]
Great PR!

**Comment by osmaczko** [comment]
waiting for tests: https://github.com/status-im/status-react/pull/13661

**Comment by 0x-r4bbit** [comment]
Nice work @osmaczko

**Comment by pavloburykh** [comment]
@osmaczko Thanx for the PR! If it is ready for QA, could you please move it to Ready for client testing column, in order it be taken into test by both desktop and mobile QA teams? Thank you!

**Comment by osmaczko** [comment]
> @osmaczko Thanx for the PR! If it is ready for QA, could you please move it to Ready for client testing column, in order it be taken into test by both desktop and QA team? Thank you!

Thanks. Ofc, done :+1:

**Comment by pavloburykh** [comment]
@osmaczko thank you for the PR. 

Tested on mobile and got a question about ENS name updating in community channel: 
### User's ENS name is updated for other community users only after ENS user posts some message in channel. Is it an expected behaviour? 

**Steps:**
1. Create a new community
2. Create a new User 1
3. Join community via newly created User 1
4. Join community via User 2 who has couple of ENS names connected
5. Try to change User's 2 ENS name (without posting new messages in channel after update) and see if User 1 can see updated ENS. 


**Actual result:** User 1 doesn't see User's 2 updated ENS until User 2 posts some message in community channel.

**Expected result: ???** User 1 sees User's 2 updated ENS name, right after it has been updated. 

PS: sorry, not sure if this problem is related to current PR. 

https://user-images.githubusercontent.com/97245802/179024412-1c8ffc4e-4ced-4d74-beac-3465dfd3125b.mp4

**Comment by osmaczko** [comment]
@pavloburykh thanks for testing. 

It seems propagating ENS name with contact identity is [not yet implemented](https://github.com/status-im/status-go/blob/530f3c7a3ac802ab502ed97692ea7a85e3492e5b/protocol/messenger.go#L982) (FYI @cammellos)

> Actual result: User 1 doesn't see User's 2 updated ENS until User 2 posts some message in community channel.

I believe the same issue happens in any other kind of chat (public, group, 1on1), could you please check?

> **Expected result: ???** User 1 sees User's 2 updated ENS name, right after it has been updated.

Yes, I would expect so.

---

IMO the issue is not related to this PR and should be solved separately. Since on mobile, display names are not implemented yet (:question:), I'm attaching a video where display names are propagated to community members as expected:

https://user-images.githubusercontent.com/33099791/179498091-310cf46d-c69f-4326-8258-1328121c1009.mp4

Changing user images also works as expected:

https://user-images.githubusercontent.com/33099791/179498947-9832506a-f27c-4f45-80c0-efccf7f00082.mp4

**Comment by pavloburykh** [comment]
@osmaczko thank you! You're right, this issue is not related to current PR. Therefore, currently there are no issues from mobile side. Waiting for desktop QA team to test the PR, after that it will be ready for merge.

**Comment by anastasiyaig** [comment]
@osmaczko 1 question i want to clarify: as user2 , having custom picture, i joined community and this user's picture is not displayed  in members list of community, or Online users section:


Screenshot 2022-07-20 at 10 55 58

I created https://github.com/status-im/status-desktop/pull/6530 to test the commit in go, double checked - it points to correct version

**Comment by anastasiyaig** [comment]
okay , the issue is a separate one related to settings

**Pull request merged**